mobile/src/main/java/org/openhab/habdroid/background/BackgroundTasksBroadcastReceiver.java

```diff
@@ -0,0 +1,107 @@
+package org.openhab.habdroid.background;
+
+import android.app.AlarmManager;
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.Log;
+import androidx.work.Constraints;
+import androidx.work.NetworkType;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+
+import org.openhab.habdroid.R;
+
+import static android.content.Context.NOTIFICATION_SERVICE;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_ID_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND;
+import static org.openhab.habdroid.background.BackgroundUtils.NOTIFICATION_TAG_BACKGROUND_ERROR;
+import static org.openhab.habdroid.background.BackgroundUtils.WORKER_TAG_SEND_ALARM_CLOCK;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ENABLED;
+import static org.openhab.habdroid.util.Constants.PREFERENCE_ALARM_CLOCK_ITEM;
+
+public class BackgroundTasksBroadcastReceiver extends BroadcastReceiver {
+    private static final String TAG = BackgroundTasksBroadcastReceiver.class.getSimpleName();
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        Log.d(TAG, "onReceive() with intent " + intent.getAction());
+
+        if (AlarmManager.ACTION_NEXT_ALARM_CLOCK_CHANGED.equals(intent.getAction())) {
+            Log.d(TAG, "Alarm clock changed");
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+                startAlarmChangedWorker(context);
+            }
+        } else if (Intent.ACTION_LOCALE_CHANGED.equals(intent.getAction())) {
+            Log.d(TAG, "Locale changed, recreate notification channels");
+            BackgroundUtils.createNotificationChannels(context);
+        }
+    }
+
+    public static void startAlarmChangedWorker(Context context) {
+        final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
+        startAlarmChangedWorker(context, prefs.getString(PREFERENCE_ALARM_CLOCK_ITEM, ""),
+                prefs.getBoolean(PREFERENCE_ALARM_CLOCK_ENABLED, false), true);
+    }
+
+    public static void startAlarmChangedWorker(Context context, String itemName, boolean enabled,
+            boolean showItemEmptyNotification) {
```

Comment:
How about we kill this parameter (and notification) and just prevent the user from clearing the item name by not allowing pref saving in that case?

Requires subclassing EditTextPreference, adding a `TextWatcher` to its EditText and graying out the OK button in that case - `UrlInputPreference` gives an example of that. I'm still in favor of a combined pref, though ;-) But I guess we can refactor that in a separate PR.

Reply:
> prevent the user from clearing the item name by not allowing pref saving in that case?

Yes, that's a good idea. I should also catch whitespace and newline from the item name.